libs/pavex/src/server/worker.rs

```diff
@@ -218,14 +224,29 @@ where
 
     fn handle_connection(
         connection: TcpStream,
-        handler: fn(http::Request<hyper::body::Incoming>, ApplicationState) -> HandlerFuture,
+        handler: fn(
+            http::Request<hyper::body::Incoming>,
+            ConnectionInfo,
+            ApplicationState,
+        ) -> HandlerFuture,
         application_state: ApplicationState,
     ) {
+        let local_addr = connection.local_addr().expect("todo");
```

Comment:
There's a problem with removing these `expect`s.

In doing so we need to either:

1. Change the function signature of handler to accommodate the error cases and then `?` them.
2. Store `Result<SocketAddr, io::Error>` in the `ConnectInfo` and make the accessors return `io::Result<SocketAddr>` just like [`TcpStream::peer_addr`](https://docs.rs/tokio/latest/tokio/net/struct.TcpStream.html#method.peer_addr).

The problem with (2) is that `io::Error` and `TcpStream` are _not_ `Clone`. So we can't `clone` + `move` either of them into the `service_fn` closure. Perhaps `ConnectInfo` needs to store a shareable handle to the `TcpStream`?

Reply:
At least in the case of HTTP3, we need a shareable handle to a connection object, contained within the `ConnectInfo` in order to allow the user to query the dynamic remote IP.

Reply:
We need to approach this differently: we don't want to extract a `peer_addr` from `TcpStream`, we need to capture the one that's returned by `accept` (https://github.com/LukeMathWalker/pavex/blob/0655340589f9b0aa62a848e9e6b460ce74c1fe87/libs/pavex/src/server/server_handle.rs#L220). We always have that for HTTP 1.x and HTTP 2.

Reply:
HTTP/3 is quite different (the peer addr can change, we have a connection id as well), so I wonder if we should just do two different enum variants.

Reply:
> we need to capture the one that's returned by `accept`.

Had a stab at that https://github.com/LukeMathWalker/pavex/pull/169/commits/98401865373f3083ef8e7d34301b7133011ae14e.

I've commented out `local_addr` because the problem above still exists for it.

Reply:
`local_addr` is not super important, I think we can leave it out for now.
We could handle it by taking care of the error ourselves, putting an `Option` inside `ConnectionInfo` rather than a `Result`, side-stepping the clonability issue.

Reply:
Removed `local_addr` entirely for now.